wapitiCore/net/crawler.py

```diff
@@ -1016,6 +1017,93 @@ def __init__(
+            page = self.get(req, follow_redirects=True, headers=head)
+            forms = []
+            for i, forma in enumerate(page.iter_forms(autofill=False)):
+                forms.append(forma)
+            form = forms[0]
+            post_params = form.post_params
+            for i, post_param_tuple in enumerate(post_params):
+                field, value = post_param_tuple
+                if value:
+                    post_params[i] = [field, value]
+                elif "password" in field:
+                    post_params[i] = [field, password]
+                elif any(ext in field for ext in list):
+                    post_params[i] = [field, username]
+
+            if form.file_name:
+                request = Request(form.url, post_params=post_params, link_depth=form.link_depth)
+            else:
+                request = Request(page._response.url, post_params=post_params, link_depth=form.link_depth)
+
+            response = self.post(request, follow_redirects=True,headers=head)
+
+        else:
+
+            page = self.get(web.Request(url), follow_redirects=True, headers=head, verify=False)
+            forms = []
+
+            for i, form in enumerate(page.iter_forms(autofill=False)):
+                if i == 0:
+                    print('')
+                    print(_("Choose the form you want to use or enter 'q' to leave :"))
+                print("{0}) {1}".format(i, form))
+                forms.append(form)
+
+            valid_choice_done = False
+            if forms:
+                nchoice = -1
+                print('')
+                while not valid_choice_done:
+                    choice = input(_("Enter a number : "))
+                    if choice.isdigit():
+                        nchoice = int(choice)
+                        if len(forms) > nchoice >= 0:
+                            valid_choice_done = True
+                    elif choice == 'q':
+                        break
+
+                if valid_choice_done:
+                    form = forms[nchoice]
+                    print('')
+                    print(_("Please enter values for the following form: "))
+                    print(_("url = {0}").format(form.url))
+
+                    post_params = form.post_params
+                    for i, post_param_tuple in enumerate(post_params):
+                        field, value = post_param_tuple
+                        if value:
+                            new_value = input(field + " (" + value + ") : ")
+                        else:
+                            new_value = input("{}: ".format(field))
+                        post_params[i] = [field, new_value]
+
+                    request = Request(page.url, post_params=post_params, link_depth=form.link_depth)
+                    response = self.post(request, follow_redirects=True)
+
+        return response
+
+    def set_session(self):
+
+        url = self._base.url
+        res = self.get_auth_form(url)
```

Comment:
I wasn't able to make it work on `testphp.vulnweb.com` without modifying the code as

- the website use `uname` for username field and `pass` for password field (with -a `test%test` option)
- without specifying `-a` it find the correct form and fill it correctly but you are giving no choice for the base URL that is used to define the scope of the scanner so it won't do the scan if the user specify a full login url with the default scope (`folder`). This is the case with the acunetix website using `http://testphp.vulnweb.com/login.php` as login page.